### ISSUE TITLE
[Bug]: [v0.26rc1][GLM5.2] 回退megamoe后MTP接受步长显著下降

### ISSUE THREAD

**Issue by MarinaMiao** [vllm-project/vllm-ascend#16195]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Your output of above commands here
```

</details>

### 🐛 Describe the bug

关闭 MegaMoE 后 MTP 接受率显著下降,导致 TPOT 升高(单步 ITL 不变)

## 现象

在同一组硬件 + 同一 PD 分离服务配置下,**仅切换 MegaMoE 开关**(通过 PR #15077 在 `use_cann_megamoe()` 函数体首行加 `return False`),观察到:

- **ITL(单步 token 间隔)几乎不变**(开启 ≈94 ms / 关闭 ≈95 ms)
- **TPOT(per-output-token 延时)显著升高**(开启 ≈37 ms / 关闭 ≈50 ms,**差异 +35%**)
- MTP 平均接受长度(Mean acceptance length)从 ~4 跌到 ~1(接受率从 ~60% 降到 ~0.2%)
- Output tok/s 下降约 5%

结论:关闭 MegaMoE 主要通过**降低 MTP 接受步长**来影响 TPOT,而不是让单步 decode 算子变慢。

## 实验环境

| 项 | 值 |
|---|---|
| 模型 | GLM-5.2-w4a8(quantization=ascend, w8a8_dynamic) |
| kv_transfer | MooncakeConnectorV1 + PR #14836(支持 P-D DCP mismatch) |
| 部署 | 4 机A3 FULL PD(P0/P1 prefill + D0/D1 decode + cond pod conductor),同超节点直接 RoCE KV 传递 |
| vllm-ascend | release_0.26.0(`release_0.26.0_0817`) |
| bench 参数 | dataset=20k_40k, cc=32, rr=1000, num=100, output-len=700 |
| MegaMoE 切换 | env `VLLM_ASCEND_ENABLE_FUSED_MC2=1` 不变;函数体 `use_cann_megamoe()` 加 `return False` 实现 toggle |

### 实验一，D节点不开DCP，关megamoe稍微劣化
P节点：dp4tp8dcp8
D节点：dp16tp2，去DCP
上下两组数据，ITL差不多，但是TPOT显著劣化 --》关闭megamoe后平均接受长度大幅减小
<img width="938" height="320" alt="Image" src="https://github.com/user-attachments/assets/af1a6521-9fa5-43ba-966f-a6973752156e" />

### 实验二，D节点开DCP，关megamoe严重劣化
P节点：dp4tp8dcp8
D节点：dp4tp8dcp8
上下两组数据，ITL差不多，但是TPOT严重劣化 --》关闭megamoe后平均接受长度接近为1
<img width="993" height="183" alt="Image" src="https://github.com/user-attachments/assets/a9fb096f-c773-41b8-a1ca-8166e74d3e8c" />

## 分析
1. **ITL(单步间隔)几乎不变** → MegaMoE 开/关并不显著影响一次 decode step 的实际耗时;这说明 TPOT 差异**不是 step 计算变慢导致**。
2. **关闭 MegaMoE 后 MoE 走 fallback `aclnnDispatchFFNCombineW4A8`**(因 `use_cann_megamoe()` 强返 False)。该路径下 w4a8 MoE 计算精度与 MegaMoE 路径存在差异,导致 **draft model 的 logits 与 main model 不一致** → draft tokens 几乎全部被 main 拒绝(接受率 ~0.2%)→ MTP 退化为单步单 token,加速效果完全失效。
## 结论 / 期望
- PR #15077 目的是**临时禁掉 MegaMoE pending bug fixes**,但实测副作用:**禁用 MegaMoE 时 MTP 基本不起作用**(接受率从 ~60% 降到 ~0.2%),致使 effective decode throughput 退回单步 decoding 速率。
- 希望社区:
  1. 评估 fallback `aclnnDispatchFFNCombineW4A8` 在 w4a8 大 EP MoE 下的精度/正确性;
  2. 或在不开 MegaMoE 时仍提供一条精度等价 fallback,以保持 MTP 接受率;
  3. 文档标注这个 trade-off,以免用户误以为关掉只是 "切算法路径、性能持平"。

**Comment by Dawn952** [comment]
@MarinaMiao 能否提供下关闭MegaMoE前后的测试文本结果，需要判断是下否是因为精度劣化导致的采信率归零。也请@aisong1988 关注下模型的特性叠加情况
